packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapHeatmapController.m

```diff
@@ -0,0 +1,159 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "GoogleMapHeatmapController.h"
+#import "JsonConversions.h"
+@import GoogleMapsUtils;
+
+@implementation FLTGoogleMapHeatmapController {
+  GMUHeatmapTileLayer *_heatmap;
+  GMSMapView *_mapView;
```

Comment:
Does the controller needs to hold reference to the mapView? It seems the only place that we use the map view is setMap, and we can make the setMap take mapView as a parameter:
```
- (void)setMap:(MapView *)mapView;
```

Reply:
Removing this from the constructor would make this class deviate from the structure of the other controllers more than I'm comfortable with without more discussion on the other issues

Reply:
Thanks! I'm ok with keeping it as is and do a refactoring later for all the files in the plugin. This change should not blocking landing the PR.